chain/chain/src/chain.rs

```diff
@@ -202,6 +227,82 @@ impl OrphanBlockPool {
     }
 }
 
+/// A wrapper for a block being processed via process_block methods.  Besides
+/// block itself and provenance it specifies whether the block has been
+/// validated.  To validate use [`BlockToProcess::check_block_validity`] method.
+struct BlockToProcess {
```

Comment:
There is actually already a data structure for this use case, called `MaybeValidated`. https://github.com/near/nearcore/blob/master/core/primitives/src/utils.rs#L34, and you can search the usage of `validate_with` to see how it is used right now.

I think it's probably better to use `MaybeValidated<Block>` instead of creating a new data structure. Additionally, `provenance` and `added` don't need to be included here so that code regarding them don't need to be modified.